src/gui/C4StartupScenSelDlg.cpp

```diff
@@ -1616,8 +1629,15 @@ void C4StartupScenSelDlg::UpdateList()
 		for (C4ScenarioListLoader::Entry *pEnt = pScenLoader->GetFirstEntry(); pEnt; pEnt = pEnt->GetNext())
 		{
 			if (pEnt->IsHidden()) continue; // no UI entry at all for hidden items
-			ScenListItem *pEntItem = new ScenListItem(pScenSelList, pEnt);
-			if (pEnt == pOldSelection) pScenSelList->SelectEntry(pEntItem, false);
+			if (!SLen(searchBar->GetText()) || SSearchNoCase(pEnt->GetName().getData(), searchBar->GetText()))
+			{
+				ScenListItem *pEntItem = new ScenListItem(pScenSelList, pEnt);
```

Comment:
Where is this deleted? I would highly recommend using smart pointers whenever possible in new code.
(Even if this might not be one of the location where it actually is possible)

Reply:
[Here.](https://github.com/openclonk/openclonk/blob/master/src/gui/C4StartupScenSelDlg.cpp#L1588)
Yes, smart pointers are great, but I don't see the benefit in rewriting working code just to use them.

Reply:
> Where is this deleted? I would highly recommend using smart pointers whenever possible in new code.
I'm usually hugely in favor of using more smart pointers, but this isn't new code. Also whoever wants to use smart pointers in C4Gui probably needs to start at the top of the object hierarchy.